src/app/app.component.ts

```diff
@@ -1,15 +1,20 @@
 import { Component, OnInit } from '@angular/core';
-import { ChampionRepositoryService } from './data/repository/champion/champion-repository.service';
+import { environment } from 'src/environments/environment';
+import ServiceFactoryHttp from './infra/factory/ServiceFactoryHttp';
+import AxiosAdapter from './infra/http/AxiosAdapter';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss'],
 })
 export class AppComponent implements OnInit {
-  constructor(private championService: ChampionRepositoryService) {}
-
-  ngOnInit() {
-    this.championService.getAll().subscribe();
+  async ngOnInit() {
```

Comment:
What do you think about removing the async from ngOnInit, and putting it like this
``` 
(async () => {
  const serviceFactory = new ServiceFactoryHttp(
    new AxiosAdapter(),
    environment.serverUrl
  );
    const championService = serviceFactory.createChampionService();
    const champions = await championService.list();
 })();
```